src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -606,6 +629,35 @@ private static bool TryParseStatusCode(string responseHeader, int statusCodeStar
             return true;
         }
 
+        private static void HandleRedirectLocationHeader(EasyRequest state, string locationValue)
+        {
+            Debug.Assert(state._isRedirect);
+            Debug.Assert(state._handler.AutomaticRedirection);
+
+            string location = locationValue.Trim();
+            //only for absolute redirects
+            if (location.StartsWith(UriSchemeHttp) || location.StartsWith(UriSchemeHttps))
+            {
+                Uri forwardUri = new Uri(location);
```

Comment:
I'm not sure what the different HTTP stacks do in this case.  I.e. for .NET Desktop, vs. Wininet vs. WinHTTP etc.  I'm not sure they throw exceptions but they might simply not follow the redirect.  We would need to test the stacks and determine the behaviors and then select the best behavior we think is good.